core/utils/utils.go

```diff
@@ -105,6 +107,29 @@ func NewBytes32ID() string {
 	return strings.Replace(uuid.NewV4().String(), "-", "", -1)
 }
 
+// NewBytes32Secret returns a randomly generated string which conforms to
+// Ethereum bytes32.
+//
+// Panics on failed attempts to read from system's PRNG.
+func NewBytes32Secret() string {
+	return NewSecret(32)
+}
+
+// NewSecret returns a new secret of the given string length.
+//
+// Panics on failed attempts to read from system's PRNG.
+func NewSecret(strLen int) string {
+	if strLen%4 != 0 {
+		panic("length must be a multiple of 4")
+	}
+	b := make([]byte, strLen*3/4)
```

Comment:
I'd actually prefer to describe this in terms of the entropy length rather than the outputted string length.

Reply:
I'd prefer that as well.  Also, what about changing NewBytes32Secret to have 32 bytes of entropy?  It may require adding a new type though.